src/classes/CellsConfig.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+//******************************************************************************
+//                              CellsConfig.php
+//
+// Author(s): Arnaud Charleroy
+// SILEX version 1.0
+// Copyright © - INRA - 2018
+// Creation date: 26 janv. 2018
+// Contact: arnaud.charleroy@inra.fr, anne.tireau@inra.fr, pascal.neveu@inra.fr
+// Last modification date:  26 janv. 2018
+// Subject: A class which represents cells configuration
+//******************************************************************************
+
+/**
+ * @link http://www.inra.fr/
+ * @copyright Copyright © INRA - 2018
+ * @license https://www.gnu.org/licenses/agpl-3.0.fr.html AGPL-3.0
+ */
+
+namespace openSILEX\handsontablePHP\classes;
+
+use \openSILEX\handsontablePHP\classes\CellConfigDefinition;
+use \openSILEX\handsontablePHP\tools\JavascriptFormatter;
+
+/**
+ * Represents a set of cell configuration unit or a function for cells configuration
+ * @see https://docs.handsontable.com/latest/Options.html#cells
+ * @see https://docs.handsontable.com/latest/Options.html#cell
+ * @author Arnaud Charleroy <arnaud.charleroy@inra.fr>
+ * @since 1.0
+ */
+class CellsConfig implements \JsonSerializable
+{
+
+    /**
+     * List of the cell definition that will be specified
+     * @var mixed array of CellConfigDefinition or Closure object
+     */
+    protected $cellsConfig = null;
+
+    /**
+     * Define if the attribute is cell or cells
+     * @var int number of the mode used
+     */
+    protected $cellMode = null;
+
+    /**
+     * Used to save information about cell attribute
+     */
+    const CELL_MODE = 0;
+    
+    /**
+     * Used to save information about cells attribute
+     */
+    const CELLS_MODE = 1;
+
+    public function __construct($cellsConfig, $cellMode = CellsConfig::CELLS_MODE)
+    {
+        $this->cellsConfig = $cellsConfig;
+        $this->cellMode = $cellMode;
+    }
+
+    /**
+     * @example
```

Comment:
can you explain what is the method doing ? The example is great but i am not sure to really understand the method action only with the example

Reply:
JsonSerialize it's an abstract method provide by \JsonSerializable interface, it's already documented . 
```php
/**
	 * (PHP 5 &gt;= 5.4.0, PHP 7)<br/>
	 * Specify data which should be serialized to JSON
	 * @link http://php.net/manual/en/jsonserializable.jsonserialize.php
	 * @return mixed data which can be serialized by <b>json_encode</b>,
	 * which is a value of any type other than a resource.
	 */
	abstract public function jsonSerialize();
}
```
Should I put a link to this description ? Like @see \JsonSerializable